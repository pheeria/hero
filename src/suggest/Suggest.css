/**
 * The geosuggest module
 * NOTE: duplicated font-sizes' are for browsers which don't support rem (only IE 8)
 */
.geosuggest {
  position: relative;
  width: 100%;
  margin: 16px auto;
  text-align: left;
  line-height: 1.5;
  font-size: 100%;
}
.geosuggest__input {
  width: 100%;
  height: 32px;
  color: rgba(0, 0, 0, 0.65);
  border: 1px solid #d9d9d9;
  border-radius: 4px;
  padding: 0 1em;
  padding: 4px 11px;
  -webkit-transition: border 0.2s, box-shadow 0.2s;
  transition: border 0.2s, box-shadow 0.2s;
}
.geosuggest__input:focus {
  border-color: #267dc0;
  box-shadow: 0 0 0 transparent;
}
.geosuggest__suggests {
  position: absolute;
  top: 100%;
  left: 0;
  right: 0;
  max-height: 25em;
  padding: 0;
  margin-top: -1px;
  background: #fff;
  border: 2px solid #267dc0;
  border-top-width: 0;
  overflow-x: hidden;
  overflow-y: auto;
  list-style: none;
  z-index: 5;
  -webkit-transition: max-height 0.2s, border 0.2s;
  transition: max-height 0.2s, border 0.2s;
}
.geosuggest__suggests--hidden {
  max-height: 0;
  overflow: hidden;
  border-width: 0;
}

/**
 * A geosuggest item
 */
.geosuggest__item {
  font-size: 18px;
  font-size: 1rem;
  padding: 0.5em 0.65em;
  cursor: pointer;
}
.geosuggest__item:hover,
.geosuggest__item:focus {
  background: #f5f5f5;
}
.geosuggest__item--active {
  background: #267dc0;
  color: #fff;
}
.geosuggest__item--active:hover,
.geosuggest__item--active:focus {
  background: #ccc;
}
.geosuggest__item__matched-text {
  font-weight: bold;
}
