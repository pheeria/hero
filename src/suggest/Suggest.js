import React from 'react';
import Geosuggest from 'react-geosuggest';
import './Suggest.css';

const Suggest = ({ city, setTempLocation }) => (
  <Geosuggest
    placeholder="Anywhere"
    onSuggestSelect={suggested => {
      if (suggested) {
        setTempLocation(suggested.location);
      }
    }}
    location={
      // eslint-disable-next-line no-undef
      new google.maps.LatLng(city.lat, city.lng)
    }
    radius="20"
  />
);

export default Suggest;
