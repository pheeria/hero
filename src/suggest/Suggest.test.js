import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Suggest from './Suggest';

describe('Suggest', () => {
  it('should render without crashing', () => {
    const tempLocation = {
      lat: 52.47732379999999,
      lng: 13.425160300000016
    };

    const wrapper = shallow(
      <Suggest setTempLocation={jest.fn()} city={tempLocation} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
