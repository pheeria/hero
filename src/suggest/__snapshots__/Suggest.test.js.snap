// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggest should render without crashing 1`] = `
<default_1$1
  autoActivateFirstSuggest={false}
  disabled={false}
  fixtures={Array []}
  getSuggestLabel={[Function]}
  highlightMatch={true}
  ignoreEnter={false}
  ignoreTab={false}
  initialValue=""
  location={mockConstructor {}}
  maxFixtures={10}
  minLength={1}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onSuggestSelect={[Function]}
  placeholder="Anywhere"
  queryDelay={250}
  radius={20}
  skipSuggest={[Function]}
  style={Object {}}
/>
`;
